src/journal.rs

```diff
@@ -225,6 +226,38 @@ impl Journal {
         sd_try!(ffi::sd_journal_get_realtime_usec(self.j, &mut timestamp_us));
         Ok(system_time_from_realtime_usec(timestamp_us))
     }
+
+    /// Adds a match by which to filter the entries of the journal.
+    /// If a match is applied, only entries with this field set will be iterated.
+    pub fn match_add<T: Into<Vec<u8>>>(&self, key: &str, val: T) -> Result<&Journal> {
```

Comment:
That seems fine to me (and we probably should at some point adjust the existing methods to do the same).